cirq-web/cirq_ts/src/bloch_sphere/components/scene.ts

```diff
@@ -0,0 +1,137 @@
+   * camera - the Three.js camera object responsible for seeing the
+   *  visualization from different locations
+   * renderer - the Three.js renderer object resposible for rendering
+   *  the visualization into the browser. We will be using the WebGL
+   * renderer
+   * controls - the Three.js OrbitControls object reponsible for managing
+   *  mouse input and moving the camera, zooming in and out, etc.
+   * containerId - the id of the HTML container (<div>, <span>, etc.)
+   *  that will contain the scene output
+   */
+  camera: Camera;
+  renderer: WebGLRenderer;
+  controls: OrbitControls;
+  containerId: string;
+
+  /**
+   * Initializes a 3D Scene proportional to the bloch sphere visualzation.
+   * @param containerId The id of the HTML div that will contain the scene output
+   * @param fov The vertical field of view for the Scene's Perspective Camera
+   * @param aspect The aspect ratio for the Scene's Perspective Camera
+   * @param near The near plane for the Scene's Perspective Camera
+   * @param far The far plane for the Scene's Perspective Camera
+   */
+  public constructor(
+    containerId = 'container',
+    fov = 75,
+    aspect: number = BlochSphereScene.VIZ_HEIGHT / BlochSphereScene.VIZ_WIDTH,
+    near = 0.1,
+    far = 1000
+  ) {
+    this.containerId = containerId;
+    this.scene = new Scene();
+    this.camera = new PerspectiveCamera(fov, aspect, near, far);
+    this.renderer = new WebGLRenderer({alpha: true});
+    this.renderer.setSize(
+      BlochSphereScene.VIZ_WIDTH,
+      BlochSphereScene.VIZ_HEIGHT
+    );
+    this.controls = new OrbitControls(this.camera, this.renderer.domElement);
+
+    this.init();
+  }
+
+  /**
+   * Initialization helper function. Also sets the starting
+   * position of the camera.
+   */
+  private init() {
+    this.camera.position.x = 6;
+    this.camera.position.y = 2;
+    this.camera.position.z = 2;
+
+    this.setUpControls();
+    this.addSceneToHTML();
+    this.setRenderSize();
+    this.animate();
+  }
+
+  /**
+   * Handles setting up the controls for OrbitControls.
+   */
+  private setUpControls() {
+    this.controls.enableDamping = true;
+    this.controls.dampingFactor = 0.05;
+    this.controls.screenSpacePanning = false;
+    this.controls.minDistance = 10;
+    this.controls.maxDistance = 50;
+    this.controls.maxPolarAngle = Math.PI;
+  }
+
+  /**
+   * Adds the 3D Scene to the HTML according to the div id
+   * provided in the constructor.
+   */
+  private addSceneToHTML() {
+    const container = document.getElementById(this.containerId)!;
+    container.appendChild(this.renderer.domElement);
+  }
+
+  /**
+   * Configures the output canvas and the viewport
+   * to make sure that elements are rendered correctly
+   * onto the scene.
+   * @param width The desired width of the rendering canvas.
+   * @param height The desired height of the rendering canvas.
+   */
+  public setRenderSize(
+    width: number = BlochSphereScene.VIZ_WIDTH,
```

Comment:
It's a little odd to call `setRenderSize()` with no arguments since it's not clear in that context what the size will be. Consider ditching the default values unless I'm missing something?